data/moves.js

```diff
@@ -14757,10 +14757,11 @@ let BattleMovedex = {
 		beforeTurnCallback(pokemon) {
 			pokemon.addVolatile('shelltrap');
 		},
-		beforeMoveCallback(pokemon) {
+		onTry(pokemon) {
 			if (pokemon.volatiles['shelltrap'] && !pokemon.volatiles['shelltrap'].gotHit) {
+				this.attrLastMove('[still]');
```

Comment:
Is this necessary? I think `Try` runs before the animation plays.

Reply:
I copied it from Focus Punch. Shoudl that be updated as well?

Reply:
Oh that's right, it does play because 'POKEMON used MOVE!' happens, which lets the client know PP was deducted. This is fine then, my bad.